portfolio/src/main/java/com/google/sps/data/Word.java

```diff
@@ -0,0 +1,45 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import com.google.cloud.translate.Translate;
+import com.google.cloud.translate.TranslateOptions;
+import com.google.cloud.translate.Translation;
+
+/** A Public class representing a word. */
+public class Word {
+
+    private final String name;
+    private final String langCode;
+    private final String translated;
+
+    /**
+     * @param name The Word itself.
+     * @param langCode The language that the word is in.
+     */
+    public Word(String name, String langCode) {
+        this.name = name;
+        this.langCode = langCode;
+        this.translated = translateWord(name,langCode);
+    }
+
+    private String translateWord(String name, String langCode){
+        Translate translate = TranslateOptions.getDefaultInstance().getService();
+        Translation translation = translate.translate(name, Translate.TranslateOption.targetLanguage(langCode));
+        return translation.getTranslatedText();
+    }
```

Comment:
I'd recommend calling this outside of this class and adding the translated text as an additional constructor parameter to this class. Part of software design is separating your application into layers. This Word class would be part of the model layer: it represents a Word in the flashcard application with the word in english, the destination language code, and the translated text. The model should be separated from the application's business logic i.e. how it interacts with the models. 

I would consider the translation to be part of the application's business logic and so I'd recommend pulling it out to the caller. So you could would 

```
String translatedWord = translate.transate(....)
Word word = new Word("Hello", "ru", translatedWord)
```

Reply:
I believe this was addressed

Reply:
Not quite there yet. My suggestion is that the Word class shouldn't know anything about the translation service. The code that instantiates the class in the servlet knows about the "business logic" of translation and so it should be responsible for passing the translated text to this class' constructor. That allows Word to just be a model class without any logic of its own.